examples/demo-app/src/cloud-providers/aws/aws-provider.js

```diff
@@ -0,0 +1,316 @@
+   * or through loadParams used in downloadMap (commented - uncomment to use)
+   * in second case, the user has to be logged in to open the map
+   */
+  async uploadMap({mapData, options = {}}) {
+    const {isPublic} = options;
+    const {map, thumbnail} = mapData;
+    const {title, description} = map && map.info;
+    const name = title;
+    // Since we share through a map link, this could be private as well
+    const level = isPublic ? 'protected' : 'private';
+    await this._saveFile(name, 'thumbnail.png', thumbnail, level);
+    await this._saveFile(name, 'meta.json', {description}, level);
+    const mapId = await this._saveFile(name, 'map.json', map, level).then(resp => resp && resp.key);
+    // If public, url for sharing is created:
+    if (isPublic) {
+      // // Comment lines below to share url with loadParams:
+      const config = {download: false, level, expires: EXPIRE_TIME_IN_SECONDS};
+      try {
+        const urlLink = await Storage.get(mapId, config);
+        this._shareUrl = encodeURIComponent(urlLink || '');
+        return {
+          shareUrl: this.getShareUrl(true)
+        };
+      } catch (e) {
+        AwsProvider._handleError('Saving failed', e);
+      }
+      // // Uncomment lines below to share url with loadParams:
+      // this._loadParam =  {identityId, level, mapId};
+      // return {shareUrl: this.getShareUrl(true)}
+    }
+    // if not public, map is saved and private map url is created
+    this._loadParam = {...this._loadParam, level, mapId};
+    return this._loadParam;
+  }
+
+  /**
+   * Required!
+   *
+   * @param onCloudLogoutSuccess
+   */
+  async logout(onCloudLogoutSuccess) {
+    try {
+      await Auth.signOut().then(() => {
+        onCloudLogoutSuccess();
+      });
+    } catch (e) {
+      AwsProvider._handleError('Signing out failed', e);
+    }
+  }
+
+  /**
+   * Required!
+   *
+   * @returns {boolean}
+   */
+  hasPrivateStorage() {
+    return PRIVATE_STORAGE_ENABLED;
+  }
+
+  /**
+   * Required!
+   *
+   * @returns {boolean}
+   */
+  hasSharingUrl() {
+    return SHARING_ENABLED;
+  }
+
+  getAccessToken() {
+    let token = null;
+    if (window.localStorage) {
+      const key = `CognitoIdentityServiceProvider.${AWS_WEB_CLIENT_ID}`;
+      const lastAuthUserKey = `${key}.LastAuthUser`;
+      const lastAuthUser = window.localStorage.getItem(lastAuthUserKey);
+      const tokenKey = `${key}.${lastAuthUser}.accessToken`;
+      token = window.localStorage.getItem(tokenKey);
+    }
+    return Boolean(token);
+  }
+
+  getUserName() {
+    if (window.localStorage) {
+      const key = `CognitoIdentityServiceProvider.${AWS_WEB_CLIENT_ID}`;
+      const lastAuthUser = window.localStorage.getItem(`${key}.LastAuthUser`);
+      const userData = JSON.parse(window.localStorage.getItem(`${key}.${lastAuthUser}.userData`));
+      return userData
+        ? userData.UserAttributes.find(item => {
+            return item.Name === 'email';
+          }).Value
```

Comment:
If find return `undefined` you'll throw an Exception maybe you should catch that